### ISSUE TITLE
Unknown transaction.type with OTLP and GRPC Spans Created by Node.js Instrumentation

### ISSUE THREAD

**Issue by astorm** [elastic/apm-server#9299]
**APM Server version** (`apm-server version`): 8.4

**Description of the problem including expected versus actual behavior**:

OpenTelemetry GRPC spans sent to the APM Server OTLP endpoint create a derived transaction with a `transaction.type` of `unknown` when the spans originate from the Node.js OpenTelemetry GRPC instrumentation.  (GRPC spans in other OpenTelemetry language API/SDKs appear to have the correct transaction.type of `request` sent)

**Steps to reproduce**:

 1. Install and run the Elastic Forked, OTel Instrumented Online Boutique microservices-demo, pointing it at a stock Elastic Cloud instance: https://github.com/julianocosta89/opentelemetry-
 2. Observe the `APM -> Services` section

Expected Behavior: `emailservice` (a python GRPC service) and `currencyservice` (a Node.js GRPC service) should both have a transaction type of `request`

Actual Behavior: Only `emailservice` has a transaction type of request, `currencyservice` has a transaction type of `unknown`

![file](https://user-images.githubusercontent.com/184372/194227616-02602518-c130-482a-a991-de34bc79fec4.png)

**Provide logs (if relevant)**:

Not logs, but using the Console Span exporter the Node.js GRPC spans look like the following

```
{
  traceId: '76e1db1e74ed86b0e529d7ce76d683ce',
  parentId: undefined,
  name: 'grpc.hipstershop.CurrencyService/Convert',
  id: 'a3a33ad55a7cc96f',
  kind: 1,
  timestamp: 1665035192673723,
  duration: 1820,
  attributes: { 'grpc.kind': 1, 'rpc.grpc.status_code': '1' },
  status: { code: 0 },
  events: []
}
{
  traceId: '9ac218a5377d9ff2569ebbf603a160c4',
  parentId: undefined,
  name: 'grpc.hipstershop.CurrencyService/GetSupportedCurrencies',
  id: '9c9ca41e54fd117d',
  kind: 1,
  timestamp: 1665035192680413,
  duration: 715,
  attributes: { 'grpc.kind': 1, 'rpc.grpc.status_code': '1' },
  status: { code: 0 },
  events: []
}
```

My best guess is that [the code in this package fails](https://github.com/elastic/apm-server/blob/626e906f2043174152214ef77572effe0eff2a2f/internal/processor/otel/traces.go#L410-L418) to identify these spans and defaults to a type of unknown.

**Comment by axw** [comment]
The bug is with the instrumentation, and should be fixed there. https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md#common-attributes states that `rpc.system` is required.

**Comment by astorm** [comment]
> The bug is with the instrumentation, and should be fixed there. https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md#common-attributes states that rpc.system is required.

Reasonable, but why not both?

Datapoint: I looked into this a bit today and this _has_ been fixed in the upstream GRPC repository 

https://github.com/open-telemetry/opentelemetry-js/blob/experimental/v0.32.0/experimental/packages/opentelemetry-instrumentation-grpc/src/grpc-js/index.ts#L198

However -- this fix didn't land until v0.32.0 of the experimental packages which are linked to the API 1.2.x and Core 1.6.x release.  ([see the compatibility matrix](https://github.com/open-telemetry/opentelemetry-js/#compatibility-matrix)) 

This means GA releases prior to 1.2/1.6/0.32 will _not_ have this fix and will report in incorrectly to elastic -- but landing [9308](https://github.com/elastic/apm-server/pull/9308) _will_ fix this for those users when they're using the latest version of the stack.  

I know this isn't technically a "bug" -- but given that our own teams have run into it and the the fix seems pretty small it seems worth doing?
